### PR TITLE
Main Page

### PR DESCRIPTION
**Pull request by SergMo**

### CODE REVIEW

File: src/components/MainPage.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+import ArticleCard from './ArticleCard';
+import Pagination from './Pagination';
+
+
+export default function MainPage() {
+	const [articles, setArticles] = useState([]);
+	const [currentPage, setCurrentPage] = useState(1);
+	const articlesPerPage = 10;
+
+	useEffect(() => {
```
Comment (Hard-Luck) [inline src/components/MainPage.jsx:11]: You should have some sort of loading state and a way of indication that to the user

File: src/components/MainPage.jsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+			.catch((error) => {
+				console.error('Erro fetching article:', error);
+			})
+	}, [])
+
+	const indexOfLastArticle = currentPage * articlesPerPage; //10= 1*10
+	const indexOfFirstArtcile = indexOfLastArticle - articlesPerPage; //0= 10-10 
+	const currentArticles = articles.slice(indexOfFirstArtcile, indexOfLastArticle);
+
+	const handePageChange = (newPage) => {
```
Comment (Hard-Luck) [inline src/components/MainPage.jsx:27]: should this say handlePageChange not handePageChage

File: src/components/MainPage.jsx (hunk 1 above)
Comment (Hard-Luck) [reply src/components/MainPage.jsx:11]: Be aware that fetch will not give errors for 400s or 500s, you may want to switch to axios

File: src/components/MainPage.jsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+export default function MainPage() {
+	const [articles, setArticles] = useState([]);
+	const [currentPage, setCurrentPage] = useState(1);
+	const articlesPerPage = 10;
+
+	useEffect(() => {
+		fetch('https://news-project-baar.onrender.com/api/articles/')
+			.then((response) => response.json())
+			.then((data) => {
+				console.log(data);
```
Comment (Hard-Luck) [inline src/components/MainPage.jsx:15]: remove this log

**Review by Hard-Luck — COMMENTED** [review]
Good work, few small changed and considerations

File: src/components/MainPage.jsx (hunk 1 above)
Comment (SergMo) [reply src/components/MainPage.jsx:11]: done

File: src/components/MainPage.jsx (hunk 2 above)
Comment (SergMo) [reply src/components/MainPage.jsx:27]: done

File: src/components/MainPage.jsx (hunk 3 above)
Comment (SergMo) [reply src/components/MainPage.jsx:15]: done

**Pull request merged**